handlers/deleteitem/deleteItem.go

```diff
@@ -0,0 +1,60 @@
+package deleteitem
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/mount-joy/thelist-lambda/db"
+	"github.com/mount-joy/thelist-lambda/handlers/iface"
+)
+
+type deleteItem struct {
+	db db.DB
+}
+
+// New returns an instance of deleteItem satisfying the RouteHandler interface
+func New() iface.RouteHandler {
+	return &deleteItem{
+		db: db.DynamoDB(),
+	}
+}
+
+// Match returns true if this RouteHandler should handle this request
+func (g *deleteItem) Match(request events.APIGatewayV2HTTPRequest) bool {
+	var re = regexp.MustCompile(`^/lists/([\w-]+)/items/([\w-]+)/?$`)
+	return request.RequestContext.HTTP.Method == "DELETE" && re.MatchString(request.RequestContext.HTTP.Path)
+}
+
+// Handle handles this request and returns the response and status code
+func (g *deleteItem) Handle(request events.APIGatewayV2HTTPRequest) (interface{}, int) {
+	listID, itemID, err := getIDs(request.RequestContext.HTTP.Path)
+	if err != nil {
+		log.Printf("Error: %s", err.Error())
+		return nil, http.StatusBadRequest
+	}
+
+	err = g.db.DeleteItem(listID, itemID)
+	if err != nil {
```

Comment:
As before with using a switch to reduce indenting

Reply:
We ended up removing this check because of your other comment so no need 😄